### ISSUE TITLE
Conflicting types for memmove on OS X Big Sur

### ISSUE THREAD

**Issue by mrichmon** [lcn2/calc#19]
When building calc v2.12.7.1 on OS X Big Sur with conflicting type errors are raised.

Compile steps:
```
$ tar jxvf ../calc-2.12.7.1.tar.bz2
$ cd calc-2.12.7.1
$ make target=Darwin clobber
$ make target=Darwin all
forming have_unistd.h
have_unistd.h formed
forming have_stdlib.h
have_stdlib.h formed
clang -DCALC_SRC -DCUSTOM -Wall    -fPIC longbits.c -c
rm -f longbits
clang -DCALC_SRC -DCUSTOM -Wall    -fPIC longbits.o -o longbits
forming longbits.h
longbits.h formed
forming align32.h
align32.h formed
forming have_string.h
have_string.h formed
forming args.h
args.h formed
forming calcerr.h
calcerr.h formed
forming conf.h
conf.h formed
clang -DCALC_SRC -DCUSTOM -Wall    -fPIC  endian.c -c
rm -f endian
clang -DCALC_SRC -DCUSTOM -Wall    -fPIC endian.o -o endian
forming endian_calc.h
endian_calc.h formed
forming have_fpos.h
have_fpos.h formed
forming have_posscl.h
have_posscl.h formed
forming have_fpos_pos.h
have_fpos_pos.h formed
forming have_offscl.h
have_offscl.h formed
forming fposval.h
fposval.h formed
forming have_const.h
have_const.h formed
forming have_memmv.h
have_memmv.h formed
forming have_newstr.h
have_newstr.h formed
forming have_times.h
have_times.h formed
forming have_uid_t.h
have_uid_t.h formed
forming terminal.h
terminal.h formed
forming have_ustat.h
have_ustat.h formed
forming have_getsid.h
have_getsid.h formed
forming have_getpgid.h
have_getpgid.h formed
forming have_gettime.h
have_gettime.h formed
forming have_getprid.h
have_getprid.h formed
forming have_urandom.h
have_urandom.h formed
forming have_rusage.h
have_rusage.h formed
forming have_strdup.h
have_strdup.h formed
forming have_unused.h
have_unused.h formed
forming calcerr.c
calcerr.c formed
rm -f custom/.all
cd custom; make -f Makefile ALLOW_CUSTOM="-DCUSTOM" AR=ar AWK=awk BINDIR="/usr/local/bin" BLD_TYPE="calc-dynamic-only" CALC_INCDIR="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/calc" CALC_SHAREDIR="/usr/local/share/calc" CAT=cat CC=" clang " CCERR="" CCMISC="" CCOPT="-O3 -g3" CCWARN="-Wall" CC_SHARE="-fPIC" CFLAGS="-DCALC_SRC -DCUSTOM -Wall    -fPIC -O3 -g3 -I.." CHMOD=chmod CMP=cmp CO=co COMMON_ADD="" COMMON_CFLAGS="-DCALC_SRC -DCUSTOM -Wall    -I.." COMMON_LDFLAGS=" " CP=cp CUSTOMCALDIR="/usr/local/share/calc/custom" CUSTOMHELPDIR="/usr/local/share/calc/custhelp" CUSTOMINCDIR="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/calc/custom" DEBUG="-O3 -g3" DEFAULT_LIB_INSTALL_PATH="/Users/mar/build/unpack/calc-2.12.7.1:/usr/local/lib:/usr/local/lib" FMT=fmt GREP=egrep HELPDIR="/usr/local/share/calc/help" ICFLAGS="-DCALC_SRC -DCUSTOM -Wall    -fPIC -I.." ILDFLAGS=" " INCDIR="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include" LANG=C LCC="clang" LDCONFIG= LDFLAGS="  " LD_SHARE="" LIBCUSTCALC_SHLIB="-single_module -undefined dynamic_lookup -dynamiclib -install_name /usr/local/lib/libcustcalc.2.12.7.1.dylib " LIBDIR="/usr/local/lib" LN=ln MAKE=make MAKEDEPEND=makedepend MAKE_FILE=Makefile MKDIR=mkdir MV=mv PURIFY="" Q="@" RANLIB="ranlib" RM=rm RMDIR=rmdir SCRIPTDIR="/usr/local/bin/cscript" SED=sed SORT=sort T= TOP_MAKE_FILE=Makefile TOUCH=touch TRUE=true VERSION=2.12.7.1 target=Darwin all
clang  -DCALC_SRC -DCUSTOM -Wall    -fPIC -O3 -g3 -I..   -c -o custtbl.o custtbl.c
In file included from custtbl.c:30:
In file included from ../value.h:33:
In file included from ../cmath.h:32:
In file included from ../qmath.h:32:
In file included from ../zmath.h:39:
../alloc.h:88:14: error: expected parameter declarator
E_FUNC void *memmove(void *s1, CONST void *s2, MEMMOVE_SIZE_T n);
             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_string.h:70:47: note: 
      expanded from macro 'memmove'
                __builtin___memmove_chk (dest, __VA_ARGS__, __darwin_obs...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_common.h:38:63: note: 
      expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                                                              ^
In file included from custtbl.c:30:
In file included from ../value.h:33:
In file included from ../cmath.h:32:
In file included from ../qmath.h:32:
In file included from ../zmath.h:39:
../alloc.h:88:14: error: expected ')'
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_string.h:70:47: note: 
      expanded from macro 'memmove'
                __builtin___memmove_chk (dest, __VA_ARGS__, __darwin_obs...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_common.h:38:63: note: 
      expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                                                              ^
../alloc.h:88:14: note: to match this '('
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_string.h:70:47: note: 
      expanded from macro 'memmove'
                __builtin___memmove_chk (dest, __VA_ARGS__, __darwin_obs...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_common.h:38:54: note: 
      expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                                                     ^
In file included from custtbl.c:30:
In file included from ../value.h:33:
In file included from ../cmath.h:32:
In file included from ../qmath.h:32:
In file included from ../zmath.h:39:
../alloc.h:88:14: warning: type specifier missing, defaults to 'int'
      [-Wimplicit-int]
E_FUNC void *memmove(void *s1, CONST void *s2, MEMMOVE_SIZE_T n);
             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_string.h:70:47: note: 
      expanded from macro 'memmove'
                __builtin___memmove_chk (dest, __VA_ARGS__, __darwin_obs...
                                                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_common.h:38:32: note: 
      expanded from macro '__darwin_obsz0'
#define __darwin_obsz0(object) __builtin_object_size (object, 0)
                               ^
In file included from custtbl.c:30:
In file included from ../value.h:33:
In file included from ../cmath.h:32:
In file included from ../qmath.h:32:
In file included from ../zmath.h:39:
../alloc.h:88:14: error: conflicting types for '__builtin___memmove_chk'
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_string.h:70:3: note: 
      expanded from macro 'memmove'
                __builtin___memmove_chk (dest, __VA_ARGS__, __darwin_obs...
                ^
../alloc.h:88:14: note: '__builtin___memmove_chk' is a builtin with type 'void
      *(void *, const void *, unsigned long, unsigned long)'
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_string.h:70:3: note: 
      expanded from macro 'memmove'
                __builtin___memmove_chk (dest, __VA_ARGS__, __darwin_obs...
                ^
1 warning and 3 errors generated.
make[1]: *** [custtbl.o] Error 1
make: *** [custom/.all] Error 2
```

**Comment by saragiotto** [comment]
Same issue here, on Mac OS Catalina.

**Comment by lcn2** [comment]
Fixed in Calc version 2.12.7.4.  Calc now compiles under macOS 11.2.
